src/metabase/search/scoring.clj

```diff
@@ -1,3 +1,106 @@
+;; # How does search scoring work?
```

Comment:
Dumb question, but can't you put all of this stuff in the namespace docstring? IIRC Marginalia will still picc up on that stuff and treat it as Markdown

Reply:
Yeah, there's no real distinction technically, I just felt like ns docstrings have a certain style to them and the three `;;` comments I wrote were stylistically different enough to belong somewhere else.

Like, for `mb.search.scoring`, I think that comment I wrote is useful, and if I were a better person I'd write a ns docstring that was something like

<blockquote>
Computes a relevancy score for search results using the weighted average of various scorers. Scores are determined by various ways of comparing the text of the search string and the item's title or description, as well as by Metabase-specific features such as how many dashboards a card appears in or whether an item is pinned.


Get the score for a result with `score-and-result`, and efficiently get the most relevant results with `top-results`.

Some of the scorers can be tweaked with configuration in [[metabase.search.config]].

</blockquote>

(uhh, I'll make that the real docstring in a second)